src/hooks/useEnableToken.ts

```diff
@@ -29,29 +29,40 @@ export const useEnableTokens = (params: Params): Result => {
     }
   }, [])
 
-  async function enableToken(): Promise<TxResult<boolean>> {
+  async function enableToken(): Promise<TxResult<boolean> | boolean> {
     assert(!enabled, 'The token was already enabled')
     assert(isConnected, "There's no connected wallet")
 
-    setEnabling(true)
-
-    // Set the allowance
-    const contractAddress = depositApi.getContractAddress()
-    const result = await erc20Api.approve(
-      userAddress,
-      tokenAddress,
-      contractAddress,
-      ALLOWANCE_MAX_VALUE,
-      params.txOptionalParams,
-    )
-
-    if (mounted.current) {
-      // Update the state
-      setEnabled(true)
-      setEnabling(false)
-    }
+    try {
+      if (!userAddress) {
+        throw new Error('No logged in user found. Please check wallet connectivity status and try again.')
+      }
+
+      setEnabling(true)
+
+      // Set the allowance
+      const contractAddress = depositApi.getContractAddress()
+
+      const result = await erc20Api.approve(
+        userAddress,
+        tokenAddress,
+        contractAddress,
+        ALLOWANCE_MAX_VALUE,
+        params.txOptionalParams,
+      )
 
-    return result
+      if (mounted.current) {
+        // Update the state
+        setEnabled(true)
+        setEnabling(false)
+      }
+
+      return result
+    } catch (error) {
+      console.error(error)
```

Comment:
We cannot hide errors, catch and log an error is almost never the right thing to do